---
title: "DE analysis on sctransformed CROP-seq data"
author: "Yifan Zhou"
date: "5/21/2019"
output: 
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=6, fig.height=4,
  echo=TRUE, warning=FALSE, message=FALSE, comment = NA, fig.align = 'center')
```

## Load data and functions

```{r load}
library(knitr)
library(kableExtra)
library(gridExtra)
library(Matrix)
library(sctransform)
library(ggplot2)
wd <- '~/Downloads/ASoC/singlecell/'
set.seed(1000)
## Necessary data
load(paste0(wd,'data/cropseq_expression.Rd'))
vst_out <- readRDS(paste0(wd,'data/sctransformed_object.rds'))
# Got 'vst_out' from:
# sctransform::vst(gene.exp,n_genes = NULL,return_gene_attr = TRUE, return_cell_attr = TRUE)

## Necessary functions
source(paste0(wd,'github/code/qq-plot.R'))
summ_pvalues <- function(pvalues){
  library(gridExtra)
  # p values distribution histogram
  plot1 <- histogram(pvalues,col='grey',type="count",xlim=c(0,1),breaks=100, main= "p value distribution")
  # pvalues qq-plot
  plot2 <- qqunif.plot(pvalues, main="p value qq-plot")
  grid.arrange(plot1,plot2, ncol=2)
}
ttest <- function(subset,tg.cells,neg.cells){
  pval <- rep(NA,nrow(subset))
  for (i in 1:nrow(subset)){
    pval[i] = t.test(subset[i,neg.cells],
                     subset[i,tg.cells])$p.value
  }
  pval <- data.frame(gene=row.names(subset),pval=pval)
  pval$adj <- p.adjust(pval$pval,method = 'fdr')
  
  summ_pvalues(pval$pval)
  cat(sum(pval$adj<0.1),'genes passed FDR=0.1 cutoff','\n')
  tb <- pval[order(pval$pval),]
  tb[,2:3] <- signif(tb[,2:3],digits = 2)
  print(kable(tb,row.names = F) %>% kable_styling() %>%
          scroll_box(width = '100%', height = '400px'))
  return(pval)
}
ttest_loop <- function(subset,tg.cells,neg.cells,nshow){
  pval <- rep(NA,nrow(subset))
  for (i in 1:nrow(subset)){
    pval[i] = t.test(subset[i,neg.cells],
                     subset[i,tg.cells])$p.value
  }
  pval <- data.frame(gene=row.names(subset),pval=pval)
  pval$adj <- p.adjust(pval$pval,method = 'fdr')
  
  # summ_pvalues(pval$pval)
  cat(sum(pval$adj<0.1),'genes passed FDR=0.1 cutoff','\n')
  tb <- head(pval[order(pval$pval),],nshow)
  tb[,2:3] <- signif(tb[,2:3],digits = 2)
  print(kable(tb,row.names = F) %>% kable_styling(position = 'center',full_width = T))
  # return(pval)
}
```

## _sctransform_ on single-cell UMI data

### Before sctransform

1. Gene expression mean ~ variance relationship
```{r before}
orig_attr <- data.frame(mean = rowMeans(gene.exp), 
                        detection_rate = rowMeans(gene.exp > 0),
                        var = apply(gene.exp, 1, var))
ggplot(orig_attr, aes(log10(mean), log10(var))) + 
  geom_point(alpha=0.3, shape=16) + geom_density_2d(size = 0.3) +
  geom_abline(intercept = 0, slope = 1, color='red') + theme_bw()
```

For the genes, we can see that up to a mean UMI count of ~1, the mean-variance follows the 1:1 line, i.e. variance and mean are roughly equal (as expected under a Poisson model). However, genes with a higher average UMI count show overdispersion compared to Poisson.

2. Mean ~ detection-rate relationship
```{r poisson}
# add the expected detection rate under Poisson model
x = seq(from = -3, to = 2, length.out = 1000)
poisson_model <- data.frame(log_mean = x, detection_rate = 1 - dpois(0, lambda = 10^x))
ggplot(orig_attr, aes(log10(mean), detection_rate)) + geom_point(alpha=0.3, shape=16) +
  geom_line(data=poisson_model, aes(x=log_mean, y=detection_rate), color='red') +
  theme_bw()
```

We see a lower than expected detection rate in the medium expression range. However, for the highly expressed genes, the rate is very close to 1.0, suggesting that there is no zero-inflation in the counts for those genes and that zero-inflation is a result of overdispersion, rather than an independent systematic bias.


### After sctransform

`sctransform` models the expression of each gene as a negative binomial random variable with a mean that depends on other variables, such as the sequencing depth for each cell.

Each observed UMI count is transformed into a Pearson residual which can be interpreted as the number of standard deviations an observed count was away from the expected mean. 
If the model accurately describes the mean-variance relationship and the dependency of mean and latent factors, then the result should have mean zero and a stable variance across the range of expression.

The `vst` function was used to estimate model parameters and performs the variance stabilizing transformation. Here we use the default -- log10 of the total UMI counts of a cell -- as the latent variable for sequencing depth for each cell.
```
vst_out <- sctransform::vst(as.matrix(gene.exp),n_genes = NULL,return_gene_attr = TRUE, return_cell_attr = TRUE)
```

```{r}
sctransform_data <- vst_out$y
cat('Output dimension (Gene x Cell):',dim(sctransform_data))
```

1. The model parameters as a function of gene mean (geometric mean)
```{r param, fig.width=8, fig.height=4}
plot_model_pars(vst_out)
```

Internally `vst` performs Poisson regression per gene with $log(\mu)=\beta_0+\beta_1 x$, where $x$ is `log10(umi)`, and $\mu$ is the expected number of UMI counts of the given gene. The above plot shows `(Intercept)`: $\beta_0$ , `log_umi`: $\beta_1$ , and `theta`: the maximum likelihood estimate of the overdispersion parameter $\theta$ under the negative binomial model, (where the variance of a gene depends on the expected UMI counts and theta: $\mu+\frac{\mu^2}{\theta}$.) In the second step, the regularized model parameters are used to turn observed UMI counts into Pearson residuals.

2. Overall properties of the transformed data
```{r after 1}
ggplot(vst_out$gene_attr, aes(residual_mean)) + geom_histogram(binwidth=0.01) +
  xlab(label = 'residual mean') + theme_bw() + 
  theme(axis.text = element_text(size = 14),
        axis.title = element_text(size = 14,face = 'bold'))

ggplot(vst_out$gene_attr, aes(residual_variance)) + geom_histogram(binwidth=0.1) + 
  geom_vline(xintercept=1, color='red') + xlim(0, 10) +  
  xlab(label = 'residual variance') + theme_bw() +
  theme(axis.text = element_text(size = 14),
        axis.title = element_text(size = 14,face = 'bold'))
```

After transformation, the mean of the gene residuals is close to zero, and most genes have a variance around one. This suggests that overall the regularized negative binomial model is a suitable model that describes the effect of sequencing depth on UMI counts. Further, after transformation there is no relationship between gene mean and variance, as the next plot shows:

```{r after 2}
ggplot(vst_out$gene_attr, aes(log10(gmean), log10(residual_variance))) +
  geom_point(alpha=0.3, shape=16) + geom_density_2d(size = 0.3) + 
  theme_bw() + theme(axis.text = element_text(size = 14),
                     axis.title = element_text(size = 14,face = 'bold'))
```

## DE analysis using independent t-test

### gRNA target: BCL11B locus

We filter the genes to be those present in at least 20% of all cells.   

```{r DE prep, echo=FALSE, results='asis'}
# sctransform_data <- vst_out$y
gene.percent <- rowMeans(gene.exp>0)
subset.genes <- names(gene.percent)[gene.percent>0.2]
subset.sctransform_data <- sctransform_data[subset.genes,]
cat('Number of genes that passed the filtering criterion:',length(subset.genes))
```

\newline

For a given target locus, we focus on 2 groups of cells: cells that contain only  gRNAs for the target locus, and cells that only contain the negative control gRNAs.

```{r echo=FALSE, results='asis'}
glocus <- "BCL11B" # UBE2Q2P1(30): 22; pos(29): 7; PBRM1(20): 130; NGEF(13): 384; KCTD13: 1; BCL11B(59): 11.
# cat('gRNA target:',glocus)
nlocus <- colSums(exp.per.enhancer>0)
tg.cells <- colnames(exp.per.enhancer)[exp.per.enhancer[glocus,]>0 & nlocus==1]
neg.cells <- colnames(exp.per.enhancer)[exp.per.enhancer["neg",]>0 & nlocus==1]
cat('Number of targeted cells with this type of gRNA uniquely:',length(tg.cells))
cat(  '\n')
cat('Number of neg control cells:',length(neg.cells))
count.gene.exp = cbind(gene.exp[ ,tg.cells],gene.exp[ ,neg.cells])
# subset.genes <- names(gene.exp.percent)[gene.exp.percent>0.2]
```

Then, we conduct categorical regression on the Pearson residuals for cells in these 2 conditions to assess how differentially expressed each gene is.

Distribution of _p_ values from the tests:
```{r BCL11B, fig.width=8, fig.height=4, results='asis'}
BCL11B.pval <- ttest(subset.sctransform_data,tg.cells,neg.cells)
```

***

### Top DE genes detected
```{r plot genes, fig.width=8, fig.height=4}
plot_gene <- function(glocus,testgene,tg.cells,neg.cells,expression,residual){
  paired.df <- data.frame(row.names = c(tg.cells,neg.cells),
                          expression = expression,
                          residual = residual,
                          condition=c(rep(glocus,length(tg.cells)),
                                      rep('neg',length(neg.cells))))
  paired.df$condition <- relevel(as.factor(paired.df$condition),glocus)
  plot1 <- ggplot(data = paired.df, aes(expression,fill=condition)) + 
    geom_histogram(position = "dodge",bins = 20) + theme_bw() + 
    labs(title=paste('test gene:',testgene), x = 'UMI count') +
    theme(plot.title = element_text(size=14),
          axis.text=element_text(size=12), axis.title=element_text(size=14),
          legend.position = 'none') 
  plot2 <- ggplot(data = paired.df, aes(residual,fill=condition)) + 
    geom_histogram(position = "dodge",bins = 20) + theme_bw() + 
    labs(title=paste0('(',glocus,' locus)'), x = 'Pearson residual') +
    theme(plot.title = element_text(size=14),
          axis.text=element_text(size=12), axis.title=element_text(size=14),
          legend.title = element_text(size=12),
          legend.text = element_text(size=11),
          legend.position = c(0.8,0.85)) 
  grid.arrange(plot1,plot2, ncol=2)
}

for (testgene in BCL11B.pval$gene[order(BCL11B.pval$pval)][1:4]){
  test.expression = as.numeric(count.gene.exp[testgene,])
  test.residual = as.numeric(sctransform_data[testgene, c(tg.cells,neg.cells)])
  plot_gene(glocus,testgene,tg.cells,neg.cells,test.expression,test.residual)
}
```

***

### Permutation test
```{r perm, eval=FALSE, include=FALSE, fig.height=4, fig.width=8, results='asis'}
combined <- c(tg.cells,neg.cells)
Nperm <- 5
for (i in 1:Nperm){
  cat('Permutation test',i,':')
  cat('\n')
  permuted <- combined[sample(length(combined))]
  permuted.tg <- permuted[1:length(tg.cells)]
  permuted.neg <- permuted[(length(tg.cells)+1):length(combined)]
  permuted.pval <- ttest(subset.sctransform_data,permuted.tg,permuted.neg)
  cat('--------------------------------------')
  cat('\n')
}
```

### Comparison with using MASTcpm

```{r MAST, eval=FALSE, include=FALSE, fig.height=4, fig.width=8, results='asis'}
library(edgeR)
library(MAST)
DE_process <- function(gcount, ncount, filtcpm=10, filtpercent=0.2, perm=F){
  # perm true of false, if true will perform a permuted version.
  coldata <- data.frame(row.names = c(colnames(gcount),colnames(ncount)),
                        condition=c(rep('G',dim(gcount)[2]),rep('N',dim(ncount)[2])))
  countall <- cbind(gcount,ncount)
  if (perm==T){
    coldata$condition <- permute(coldata$condition)
  }
  y <- DGEList(counts= countall,group=coldata$condition)
  keep <- rowSums(edgeR::cpm(y)[,y$samples$group=="N"]>filtcpm) >= dim(ncount)[2] * filtpercent
  y <- y[keep, keep.lib.sizes=FALSE]
  return(y)
}
run_MASTcpmDetRate <- function(y) {
  countfl <- y$counts
  grp <-y$samples[,"group"]
  names(grp) <- rownames(y$samples)
  cdr <- scale(colSums(countfl > 0)/dim(countfl)[1])
  dge <- DGEList(counts = countfl)
  dge <- edgeR::calcNormFactors(dge)
  cpms <- edgeR::cpm(dge)
  sca <- FromMatrix(exprsArray = log2(cpms + 1),
                    cData = data.frame(wellKey = names(grp),
                                       grp = grp, cdr = cdr))
  zlmdata <- zlm.SingleCellAssay(~cdr + grp, sca)
  mast <- lrTest(zlmdata, "grp")
  
  # hist(mast[, "hurdle", "Pr(>Chisq)"], 50)
  df = data.frame(pval = mast[, "hurdle", "Pr(>Chisq)"],
                  row.names = names(mast[, "hurdle", "Pr(>Chisq)"]))
  df$fdr <- p.adjust(df$pval, method="BH")
  b <- getLogFC(zlmdata)
  df <- cbind(df, b[b$contrast=="grpN",c("logFC", "varLogFC","z"), with=F])
  df <- df[order(df$pval),]
  return(df)
}

count.ls = list(tg.count = gene.exp[ ,tg.cells],
                neg.count = gene.exp[ ,neg.cells])
y <- DE_process(count.ls$tg.count, count.ls$neg.count, 
                filtcpm=0, filtpercent=0.1, perm=F)
res <- run_MASTcpmDetRate(y)
summ_pvalues(res$pval)
cat(dim(res[res$fdr < 0.1,])[1],"genes passed FDR=0.1 cutoff",'\n')
tb <- signif(as.matrix(head(res)),digit=2)
print(kable(tb) %>% kable_styling(position = 'center',full_width = T))
```

***

### Other perturbed target loci
```{r, results='asis'}
enh_loci <- row.names(exp.per.enhancer)[-c(2,13,17,20,22)]
neg.cells <- colnames(exp.per.enhancer)[exp.per.enhancer["neg",]>0 & nlocus==1]
for (glocus in enh_loci){
  cat('Target locus:',glocus)
  cat('  \n')
  nlocus <- colSums(exp.per.enhancer>0)
  tg.cells <- colnames(exp.per.enhancer)[exp.per.enhancer[glocus,]>0 & nlocus==1]
  cat('Number of targeted cells:',length(tg.cells))
  cat('  \n')
  ttest_loop(subset.sctransform_data,tg.cells,neg.cells,3)
  cat('--------------------------------------')
  cat('\n')
}
```

### Positive controls
```{r, results='asis'}
pos_loci <- row.names(exp.per.enhancer)[c(17,20,22)]
# neg.cells <- colnames(exp.per.enhancer)[exp.per.enhancer["neg",]>0 & nlocus==1]
for (glocus in pos_loci){
  if (glocus=='pos'){
    cat('Positive locus:','SNAP91')
    cat('  \n')
  } else {
    cat('Positive locus:',glocus)
    cat('  \n')
  }
  nlocus <- colSums(exp.per.enhancer>0)
  tg.cells <- colnames(exp.per.enhancer)[exp.per.enhancer[glocus,]>0 & nlocus==1]
  cat('Number of targeted cells:',length(tg.cells))
  cat('  \n')
  ttest_loop(subset.sctransform_data,tg.cells,neg.cells,4)
  cat('--------------------------------------')
  cat('\n')
}
```

## Session Information
```{r session}
sessionInfo()
```


