---
title: "ashr on sctransformed CROP-seq data"
author: "Yifan Zhou"
date: "6/16/2019"
output:   
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=6, fig.height=4, fig.path='Figs/',
  echo=TRUE, warning=FALSE, message=FALSE, comment = NA, fig.align = 'center')
```

```{r load}
library(ashr)
library(ggplot2)
library(knitr)
library(kableExtra)
options(stringsAsFactors = F)
wkdir <- '~/Downloads/ASoC/singlecell/'
load(paste0(wkdir,'data/cropseq_expression.Rd'))
```

## Data property
```{r gRNA groups}
nlocus <- colSums(exp.per.enhancer>0)
gene.percent <- rowMeans(gene.exp>0)
hist(gene.percent,xlab = '% of cells a gene is in',ylab = 'Count')
ncell_uniq <- rep(NA,nrow(exp.per.gRNA))
for (i in 1:nrow(exp.per.gRNA)){
  glocus <- row.names(exp.per.gRNA)[i]
  ncell_uniq[i]=sum(exp.per.gRNA[glocus,]>0 & nlocus==1)
}
tmp <- sapply(strsplit(row.names(exp.per.gRNA),split = '_'), 
              function(x){paste(x[1],x[2],sep = '_')})
names(ncell_uniq) <- tmp
cat('Number of cells uniquely targeted by each gRNA:')
print(ncell_uniq)
```

## Categorical regression on `sctransform`ed data

We transformed the single-cell raw count data using `sctransform` to Pearson residuals that supposedly follow $N(0,1)$ distribution per gene.

We filtered the genes to be those present in at least 20% of all cells (8117 in total). 

Then, for each target gRNA, we selected 2 groups of cells: cells that only contain the target gRNAs, and cells that only contain the negative control gRNAs. We conducted categorical regression on the Pearson residuals (~ 2 conditions) to obtain the summary statistics reflecting how differentially expressed each gene is under the 2 conditions. 70 (gRNA vs neg ctrl) pairs were tested.


```{r eval=FALSE}
categoric <- function(mtx,tg.cells,neg.cells){
  pval <- rep(NA,nrow(mtx))
  beta <- rep(NA,nrow(mtx))
  se <- rep(NA,nrow(mtx))
  subset <- mtx[,c(tg.cells,neg.cells)]
  for (i in 1:nrow(subset)){
    df <- data.frame(row.names = c(tg.cells,neg.cells),
                     residual = as.numeric(subset[i,]),
                     condition=c(rep(1,length(tg.cells)),
                                 rep(0,length(neg.cells))))
    cat.lm <- summary(lm(residual~factor(condition), data = df))
    beta[i] <- cat.lm$coefficients[2,1]
    se[i] <- cat.lm$coefficients[2,2]
    pval[i] <- cat.lm$coefficients[2,4]
  }
  return(list(beta=beta,se=se,pval=pval))
}
vst_out <- readRDS(paste0(wkdir,'data/sctransformed_object.rds'))
sctransform_data <- vst_out$y
## Filter the genes to be present in > 20% cells:
subset.genes <- names(gene.percent)[gene.percent>0.2]
subset.sctransform_data <- sctransform_data[subset.genes,]
dim(subset.sctransform_data)
## Categorical regression on sctransformed data 
summary_per_gene <- list()
neg.cells <- colnames(exp.per.enhancer)[exp.per.enhancer["neg",]>0 & nlocus==1]
# btw each gRNA condition and neg ctrl
for (glocus in row.names(exp.per.gRNA)[-c(51,72:76)]){
  print(paste('gRNA target:',glocus))
  tg.cells <- colnames(exp.per.gRNA)[exp.per.gRNA[glocus,]>0 & nlocus==1]
  print(paste('# of targeted cells:',length(tg.cells)))
  summary_per_gene[[glocus]] <- categoric(subset.sctransform_data,tg.cells,neg.cells)
}
saveRDS(summary_per_gene,file = 'data/categoric_summary.sctrans_gene0.2.rds')
```

## ashr meta-analysis on regression summary statistics

For each gene, we combined the 70 DE effect sizes and standard errors estimated in the previous step, and used an adaptive shrinkage method, `ashr`, to discern which effect sizes are truly non-zero (significant). Here, we used a cutoff of 0.1 for `lsfr` (the local false sign rate).
```{r load summary}
summary_per_gene <- readRDS(paste0(wkdir,'data/categoric_summary.sctrans_gene0.2.rds'))
beta.mtx <- sapply(summary_per_gene, function(x){x$beta})
se.mtx <- sapply(summary_per_gene, function(x){x$se})
pval.mtx <- sapply(summary_per_gene, function(x){x$pval})
subset.genes <- names(gene.percent)[gene.percent>0.2]
rownames(beta.mtx) <- subset.genes
rownames(se.mtx) <- subset.genes
rownames(pval.mtx) <- subset.genes
```

```{r ashr, eval=FALSE}
lfsr.mtx <- data.frame(matrix(nrow = nrow(beta.mtx),ncol = ncol(beta.mtx)),
                       row.names = rownames(beta.mtx))
names(lfsr.mtx) <- colnames(beta.mtx)
betaest.mtx <- lfsr.mtx
for (i in 1:nrow(beta.mtx)){
  ash.tmp <- ash(beta.mtx[i,],se.mtx[i,])
  lfsr.mtx[i,] <- get_lfsr(ash.tmp)
  betaest.mtx[i,] <- get_pm(ash.tmp)
}
save(betaest.mtx,lfsr.mtx,file = 'data/ashr_estimation.Rdata')
```

```{r}
load(paste0(wkdir,'data/ashr_estimation.Rdata'))
tmp.gRNA <- colSums(lfsr.mtx<0.1)
tmp.gene <- rowSums(lfsr.mtx<0.1)
signif_genes <- row.names(betaest.mtx)[tmp.gene>0]
cat('There are',length(signif_genes),'significant genes in total.')
```

### gRNA targets per significant gene

Here are all the significant genes ($lfsr<0.1$) and the gRNA groups where they are differentially expressed:
```{r gene-gRNA}
signif_genes.gRNA.lst <- list()
for(i in signif_genes){
  gRNA.group <- colnames(betaest.mtx)[lfsr.mtx[i,]<0.1]
  signif_genes.gRNA.lst[[i]] <- sapply(strsplit(gRNA.group,split = '_'), 
                                       function(x){paste(x[1],x[2],sep = '_')})
}
# Make table:
compact.genes.lst <- sapply(signif_genes.gRNA.lst,FUN = function(x){paste(x,collapse = ', ')})
compact.genes.df <- as.data.frame(compact.genes.lst)
tmp.indx <- order(rownames(compact.genes.df))
compact.genes.df <- data.frame(signif_gene=rownames(compact.genes.df)[tmp.indx],
                               gRNA_targets=compact.genes.df[tmp.indx,])
# Present table:
kable(compact.genes.df) %>%
  kable_styling() %>%
  scroll_box(width = "100%", height = "400px")

num_gRNA_pergene <- sapply(signif_genes.gRNA.lst,FUN = function(x){length(x)})
hist(num_gRNA_pergene,xlab = 'Number of gRNA targets per gene',ylab = 'Count',
     main = 'Distribution of 462 significant genes')
```

Examples of `ashr` shrinkage on estimated effect sizes of selected genes:

* Genes with high frequency of appearance
```{r fig.height=6, fig.width=8}
library(gridExtra)
plot_beta <- function(gene){
  num <- num_gRNA_pergene[gene]
  log10_lfsr <- log10(as.numeric(lfsr.mtx[gene,]))
  beta.plot <- qplot(as.numeric(beta.mtx[gene,]),as.numeric(betaest.mtx[gene,]),
                     color=log10_lfsr, main = paste0(gene,' (',num,')'),
                     xlab = 'beta from regression',
                     ylab = 'beta after shrinkage') + 
               geom_abline(intercept = 0,slope = 1,linetype = "dotted")
  return(beta.plot)
}

plot1 <- plot_beta('ARAF')
plot2 <- plot_beta('CWF19L2')
plot3 <- plot_beta('THG1L')
plot4 <- plot_beta('ZNF48')
grid.arrange(plot1,plot2,plot3,plot4,ncol=2)
```

* Genes with low frequency of appearance
```{r, fig.height=6,fig.width=8}
plot1 <- plot_beta('A2M')
plot2 <- plot_beta('SUMO3')
plot3 <- plot_beta('DNAJC9')
plot4 <- plot_beta('APC')
grid.arrange(plot1,plot2,plot3,plot4,ncol=2)
```

Examples of regression _z_ score distributions for selected genes:

* Genes with high frequency of appearance
```{r zscore1, fig.height=6, fig.width=8}
plot_zscore <- function(gene){
  num <- num_gRNA_pergene[gene]
  zscore <- as.numeric(beta.mtx[gene,])/as.numeric(se.mtx[gene,])
  zscore.df <- data.frame(score=zscore)
  zscore.plot <- ggplot(zscore.df,aes(zscore)) + geom_histogram(aes(y=..density..),bins = 20) +
    labs(x = 'z-score from regression',y='frequency',title = paste0(gene,' (',num,')')) +
    stat_function(fun = dnorm, args = list(mean = 0, sd = 1),col=2)
  return(zscore.plot)
}
plot1 <- plot_zscore('ARAF')
plot2 <- plot_zscore('CWF19L2')
plot3 <- plot_zscore('THG1L')
plot4 <- plot_zscore('ZNF48')
grid.arrange(plot1,plot2,plot3,plot4,ncol=2)
```

* Genes with low frequency of appearance
```{r zscore2, fig.height=6, fig.width=8}
plot1 <- plot_zscore('A2M')
plot2 <- plot_zscore('SUMO3')
plot3 <- plot_zscore('DNAJC9')
plot4 <- plot_zscore('APC')
grid.arrange(plot1,plot2,plot3,plot4,ncol=2)
```

```{r eval=FALSE, include=FALSE}
source(paste0(wkdir,'github/code/qq-plot.R'))
summ_pvalues <- function(pvalues){
  library(gridExtra)
  # p values distribution histogram
  plot1 <- histogram(pvalues,col='grey',type="count",xlim=c(0,1),breaks=100, main= "p value distribution")
  # pvalues qq-plot
  plot2 <- qqunif.plot(pvalues, main="p value qq-plot")
  grid.arrange(plot1,plot2, ncol=2)
}
summ_pvalues(as.numeric(pval.mtx['ARAF',]))
```


### Significant genes per gRNA target
```{r gRNA-gene}
gRNA.signif_genes.lst <- list()
for (i in signif_genes){
  cells <- signif_genes.gRNA.lst[[i]]
  for (c in cells){
    if (is.na(match(c, names(gRNA.signif_genes.lst)))){
      gRNA.signif_genes.lst[[c]] <- i
    } else {
      gRNA.signif_genes.lst[[c]] <- c(gRNA.signif_genes.lst[[c]],i)
    }
  }
}
# Make table:
compact.gRNAs.lst <- sapply(gRNA.signif_genes.lst,FUN = function(x){paste(x,collapse = ', ')})
compact.gRNAs.df <- as.data.frame(compact.gRNAs.lst)
tmp.indx <- order(rownames(compact.gRNAs.df))
compact.gRNAs.df <- data.frame(gRNA_target=rownames(compact.gRNAs.df)[tmp.indx],
                               signif_genes=compact.gRNAs.df[tmp.indx,])
compact.gRNAs.df$gRNA_target <- sapply(strsplit(compact.gRNAs.df$gRNA_target,split = '_'), function(x){paste(x[1],x[2],sep = '_')})
compact.gRNAs.df$num_cells <- ncell_uniq[compact.gRNAs.df$gRNA_target]
compact.gRNAs.df <- compact.gRNAs.df[,c(3,1,2)]
# Present table:
kable(compact.gRNAs.df) %>%
  kable_styling() %>%
  scroll_box(width = "100%", height = "400px")

num_gene_pergRNA <- sapply(gRNA.signif_genes.lst,FUN = function(x){length(x)})
hist(num_gene_pergRNA,xlab = 'Number of DE genes per gRNA target',ylab = 'Count',
     main = 'Distribution of 70 gRNAs')
```

### GO over-representation analysis

Gene ontology enrichment of the 462 significant genes (out of ~30,000 genes in total):

* Enrichment based on biological processes

```{r GO signif_genes, fig.width=8, fig.height=3}
gene_info <- read.delim(paste0(wkdir,'data/genes.tsv'),header = F,
                        sep = '\t',col.names = c('ID','name'))
gene_info$ID <- sapply(strsplit(gene_info$ID,split = '[.]'),
                       FUN = function(x){x[[1]]})
tmp.indx <- match(signif_genes,gene_info$name)
ensembl.signif_genes <- gene_info$ID[tmp.indx]

library(org.Hs.eg.db)
library(clusterProfiler)
ego.bp.signif_genes <- enrichGO(gene       = ensembl.signif_genes,
                                universe      = gene_info$ID,
                                OrgDb         = org.Hs.eg.db,
                                keyType       = 'ENSEMBL',
                                ont           = "BP",
                                pAdjustMethod = "BH",
                                pvalueCutoff  = 0.01,
                                qvalueCutoff  = 0.05,
                                readable      = TRUE)
barplot(ego.bp.signif_genes, drop=TRUE, showCategory=6)
kable(head(ego.bp.signif_genes),row.names = F) %>%
  kable_styling() %>%
  scroll_box(width = "100%", height = "300px")
```

\newline

* Enrichment based on cellular components

```{r, fig.width=8, fig.height=3}
ego.cc.signif_genes <- enrichGO(gene       = ensembl.signif_genes,
                                universe      = gene_info$ID,
                                OrgDb         = org.Hs.eg.db,
                                keyType       = 'ENSEMBL',
                                ont           = "CC",
                                pAdjustMethod = "BH",
                                pvalueCutoff  = 0.01,
                                qvalueCutoff  = 0.05,
                                readable      = TRUE)
barplot(ego.cc.signif_genes, drop=TRUE, showCategory=6)
kable(head(ego.cc.signif_genes),row.names = F) %>%
  kable_styling() %>%
  scroll_box(width = "100%", height = "300px")
```

\newline

[Gene ontology enrichment of the 53 significant genes that appeared in > 3 gRNA conditions (out of ~30,000 genes in total) does not result in anything significant.]

### Gene set enrichment analysis

with ordered `geneList` values being the number of gRNA groups each gene is differentially expressed in.

* Enrichment based on biological processes
```{r gsea, fig.width=8, fig.height=4}
gene.num <- sapply(signif_genes.gRNA.lst,FUN = function(x){length(x)})
names(gene.num) <- ensembl.signif_genes
gene.num <- gene.num[order(gene.num,decreasing = T)]

gsea.signif_genes <- gseGO(geneList     = gene.num,
                           OrgDb        = org.Hs.eg.db,
                           keyType      = 'ENSEMBL',
                           ont          = "BP",
                           nPerm        = 1000,
                           minGSSize    = 100,
                           maxGSSize    = 500,
                           pvalueCutoff = 0.05,
                           verbose      = FALSE)
dotplot(gsea.signif_genes)
# convert the results back to gene names
convert <- function(x){
  indx <- match(x,gene_info$ID)
  names <- gene_info$name[indx]
  return(paste(names,collapse = '/'))
}
tmp <- strsplit(gsea.signif_genes@result$core_enrichment,split = '/')
tmp1 <- sapply(tmp,FUN = convert)
gsea.signif_genes@result$core_enrichment <- tmp1
kable(head(gsea.signif_genes),row.names = F) %>%
  kable_styling() %>%
  scroll_box(width = "100%", height = "400px")
```

## Permutation test

Please see [this page](https://gradonion.github.io/CROP-seq/ashr_permutation.html) for results of permutation tests.

## Session Information
```{r session}
sessionInfo()
```
