---
title: "Permutation test â€” cis-gene analysis (t-test + ash)"
author: "Yifan Zhou"
date: "6/27/2019"
output:  
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=9, fig.height=3, fig.path='Figs/',
  echo=TRUE, warning=FALSE, message=FALSE, comment = NA, fig.align = 'center')
```

```{r load data}
library(ashr)
library(ggplot2)
library(knitr)
library(kableExtra)
library(gridExtra)
options(stringsAsFactors = F)
wkdir <- '~/Downloads/ASoC/singlecell/'
# ash on ttest stats function
source(paste0(wkdir,'github/code/cis_ash_scttest.R'))
load(paste0(wkdir, 'data/full.cisgenes_SNP.expanded.Rdata'))
```

```{r, eval=F}
load(paste0(wkdir,'data/cropseq_expression.Rd'))
vst_out <- readRDS(paste0(wkdir,'data/sctransformed_object.rds'))
sctransform_data <- vst_out$y
load(paste0(wkdir, 'data/full.cisgenes_SNP.expanded.Rdata'))
# subsetting
nonzero.cells <- colnames(exp.per.gRNA)[colSums(exp.per.gRNA>0)>0]
subset.genes <- unique(full_genes.1Mb.expanded$cisGene)
subset.sctransform_data <- sctransform_data[subset.genes,nonzero.cells]
```

## Differential expression test on permutated `sctransform`ed data

We still use the gene $\times$ cell matrix holding `sctransform`ed Pearson residuals. We filter out genes expressed in < **10%** of all cells, and cells with no gRNA detected. For _cis_ analysis, we focus only on the genes within $\pm$ 200 kb/ 500 kb/ 1 MB of the SNP loci of interest.

In the original gRNA-based DE analysis, for each gRNA target (63 in total), we partitioned the cells into 2 groups: cells that contain the target gRNA , and cells that don't contain the target gRNA.

In the permutation tests here, for each gRNA target, we divide the cells into 2 groups **randomly**, while keeping the number of cells in each group the same as before. Then, we conduct categorical regression on the Pearson residuals (based on the permuted conditions) to obtain the summary statistics. As before, all 63 partitions were tested, and test statistics are combined together for subsequent `ash` shrinkage.

```{r functions, eval=F}
# categoric regression function
source('~/Downloads/ASoC/singlecell/github/code/categoric_regression.R')
# permutation function: test for each gRNA target (1 vs others, conditions permuted)
permute_categ <- function(seed){
  set.seed(seed)
  perm.summary <- list()
  for (glocus in row.names(exp.per.gRNA)[-c(72:76)]){
    print(paste('gRNA target:',glocus))
    tg.cells <- colnames(exp.per.gRNA)[exp.per.gRNA[glocus,]>0]
    tg.indx <- sample(length(nonzero.cells),length(tg.cells))
    tg.cells <- nonzero.cells[tg.indx]
    neg.cells <- nonzero.cells[-tg.indx]
    perm.summary[[glocus]] <- categoric(subset.sctransform_data,tg.cells,neg.cells)
  }
  return(perm.summary)
}
```

## Permutation 1

```{r perm1}
# summary_per_gene <- permute_categ(1952) # five random seeds: 1952 8076  966  275 7816
# saveRDS(summary_per_gene,file = paste0(wkdir,'data/cis.categ.perm.summary01.rds'))
summary_per_gene <- readRDS(paste0(wkdir,'data/cis.categ.perm.summary01.rds'))
```

### ash on 200 kb _cis_ genes
```{r perm1.200kb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.200kb.expanded,'gRNA')
```

***

### ash on 500 kb _cis_ genes
```{r perm1,500kb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.500kb.expanded,'gRNA')
```

***

### ash on 1 Mb _cis_ genes
```{r perm1.1Mb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.1Mb.expanded,'gRNA')
```

***

## Permutation 2

```{r perm2}
# summary_per_gene <- permute_categ(8076) # five random seeds: 1952 8076  966  275 7816
# saveRDS(summary_per_gene,file = paste0(wkdir,'data/cis.categ.perm.summary02.rds'))
summary_per_gene <- readRDS(paste0(wkdir,'data/cis.categ.perm.summary02.rds'))
```

### ash on 200 kb _cis_ genes
```{r perm2.200kb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.200kb.expanded,'gRNA')
```

***

### ash on 500 kb _cis_ genes
```{r perm2.500kb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.500kb.expanded,'gRNA')
```

***

## Permutation 3

```{r perm3}
# summary_per_gene <- permute_categ(966) # five random seeds: 1952 8076  966  275 7816
# saveRDS(summary_per_gene,file = paste0(wkdir,'data/cis.categ.perm.summary03.rds'))
summary_per_gene <- readRDS(paste0(wkdir,'data/cis.categ.perm.summary03.rds'))
```

### ash on 200 kb _cis_ genes
```{r perm3.200kb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.200kb.expanded,'gRNA')
```

***

### ash on 500 kb _cis_ genes
```{r perm3.500kb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.500kb.expanded,'gRNA')
```

***

## Permutation 4

```{r perm4}
# summary_per_gene <- permute_categ(275) # five random seeds: 1952 8076  966  275 7816
# saveRDS(summary_per_gene,file = paste0(wkdir,'data/cis.categ.perm.summary04.rds'))
summary_per_gene <- readRDS(paste0(wkdir,'data/cis.categ.perm.summary04.rds'))
```

### ash on 200 kb _cis_ genes
```{r perm4.200kb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.200kb.expanded,'gRNA')
```

***

### ash on 500 kb _cis_ genes
```{r perm4.500kb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.500kb.expanded,'gRNA')
```

***

## Permutation 5

```{r perm5}
summary_per_gene <- readRDS(paste0(wkdir,'data/cis.categ.perm.summary05.rds'))
```

### ash on 200 kb _cis_ genes
```{r perm5.200kb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.200kb.expanded,'gRNA')
```

***

### ash on 500 kb _cis_ genes
```{r perm5.500kb, results='asis'}
ash_on_scttest(summary_per_gene,full_genes.500kb.expanded,'gRNA')
```

***
